Sources/Db/APIs/MySQL.php

```diff
@@ -1349,6 +1349,8 @@ public function calculate_type(string $type_name, ?int $type_size = null, bool $
 			$types = [
 				'inet' => 'varbinary',
 				'uuid' => 'binary',
+				'mediumtext' => 'text',
+				'largetext' => 'text',
```

Comment:
This doesn't seem correct to me. Why would we want to change MEDIUMTEXT and LARGETEXT data types to regular TEXT?

Reply:
You're right. This means something changed somewhere, because in Alpha 1 `database.php` worked as expected.

Reply:
Now a query like this is generated:
```sql
CREATE TABLE `smf_alpha`.smf_lp_pages ( `page_id` smallint(5) unsigned auto_increment, `alias` varchar(255) NOT NULL , `description` varchar(255) , `content` mediumtext(0) NOT NULL , `type` varchar(10) NOT NULL DEFAULT 'bbc', PRIMARY KEY (page_id), UNIQUE alias (alias(191))) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci
```
And it needs to be like this:
```sql
CREATE TABLE `smf_alpha`.smf_lp_pages ( `page_id` smallint(5) unsigned auto_increment, `alias` varchar(255) NOT NULL , `description` varchar(255) , `content` mediumtext NOT NULL , `type` varchar(10) NOT NULL DEFAULT 'bbc', PRIMARY KEY (page_id), UNIQUE alias (alias(191))) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci
```

Reply:
type hinting issue is defaulting the size to 0 and it needs to be nullable so when null we don't pass a size.